File: example/android/sslcommerzlibrary/src/main/java/com/sslwireless/sslcommerzlibrary/model/initializer/SSLCShipmentInfoInitializer.java
package com.sslwireless.sslcommerzlibrary.model.initializer;

public class SSLCShipmentInfoInitializer {
    private String shipmentMethod;
    private String shipAddress2;
    private String shipState;
    private int numOfItems;

    private ShipmentDetails shipmentDetails;

    public SSLCShipmentInfoInitializer(String shipmentMethod,
                                       int numOfItems, ShipmentDetails shipmentDetails) {
        this.shipmentMethod = shipmentMethod;
        this.numOfItems = numOfItems;

        if (shipmentMethod.equalsIgnoreCase("yes"))
            this.shipmentDetails = shipmentDetails;
        else
            this.shipmentDetails = null;
    }

    public SSLCShipmentInfoInitializer addShipAddress2(String shipAddress2) {
        this.shipAddress2 = shipAddress2;
        return this;
    }

    public SSLCShipmentInfoInitializer addShipState(String shipState) {
        this.shipState = shipState;
        return this;
    }

    public String getShipmentMethod() {
        return shipmentMethod;
    }

    public void setShipmentMethod(String shipmentMethod) {
        this.shipmentMethod = shipmentMethod;
    }

    public int getNumOfItems() {
        return numOfItems;
    }

    public void setNumOfItems(int numOfItems) {
        this.numOfItems = numOfItems;
    }

    public ShipmentDetails getShipmentDetails() {
        return shipmentDetails;
    }

    public void setShipmentDetails(ShipmentDetails shipmentDetails) {
        this.shipmentDetails = shipmentDetails;
    }

    public String getShipAddress2() {
        return shipAddress2;
    }

    public void setShipAddress2(String shipAddress2) {
        this.shipAddress2 = shipAddress2;
    }

    public String getShipState() {
        return shipState;
    }

    public void setShipState(String shipState) {
        this.shipState = shipState;
    }

    public static class ShipmentDetails {
        private String shipName;
        private String shipAddress1;
        private String shipCity;
        private String shipPostCode;
        private String shipCountry;

        public ShipmentDetails(String shipName, String shipAddress1,
                               String shipCity, String shipPostCode, String shipCountry) {
            this.shipName = shipName;
            this.shipAddress1 = shipAddress1;
            this.shipCity = shipCity;
            this.shipPostCode = shipPostCode;
            this.shipCountry = shipCountry;
        }

        public String getShipName() {
            return shipName;
        }

        public void setShipName(String shipName) {
            this.shipName = shipName;
        }

        public String getShipAddress1() {
            return shipAddress1;
        }

        public void setShipAddress1(String shipAddress1) {
            this.shipAddress1 = shipAddress1;
        }

        public String getShipCity() {
            return shipCity;
        }

        public void setShipCity(String shipCity) {
            this.shipCity = shipCity;
        }

        public String getShipPostCode() {
            return shipPostCode;
        }

        public void setShipPostCode(String shipPostCode) {
            this.shipPostCode = shipPostCode;
        }

        public String getShipCountry() {
            return shipCountry;
        }

        public void setShipCountry(String shipCountry) {
            this.shipCountry = shipCountry;
        }
    }
}

File: lib/model/SSLCShipmentInfoInitializer.dart
import 'package:flutter/cupertino.dart';
import 'package:json_annotation/json_annotation.dart';
part 'SSLCShipmentInfoInitializer.g.dart';
@JsonSerializable()
class SSLCShipmentInfoInitializer {
  String shipmentMethod;
  String shipAddress2;
  String shipState;
  int numOfItems;
  ShipmentDetails shipmentDetails;

  SSLCShipmentInfoInitializer({@required this.shipmentMethod, @required this.numOfItems,
      @required this.shipmentDetails});

  Map<String, dynamic> toJson() => _$SSLCShipmentInfoInitializerToJson(this);
  factory SSLCShipmentInfoInitializer.fromJson(Map<String, dynamic> json) => _$SSLCShipmentInfoInitializerFromJson(json);

  SSLCShipmentInfoInitializer addShipAddress2(String shipAddress2) {
    this.shipAddress2 = shipAddress2;
    return this;
  }

  SSLCShipmentInfoInitializer addShipState(String shipState) {
    this.shipState = shipState;
    return this;
  }

  String getShipmentMethod() {
    return this.shipmentMethod;
  }

  void setShipmentMethod(String shipmentMethod) {
    this.shipmentMethod = shipmentMethod;
  }

  int getNumOfItems() {
    return this.numOfItems;
  }

  void setNumOfItems(int numOfItems) {
    this.numOfItems = numOfItems;
  }

  ShipmentDetails getShipmentDetails() {
    return this.shipmentDetails;
  }

  void setShipmentDetails(
      ShipmentDetails shipmentDetails) {
    this.shipmentDetails = shipmentDetails;
  }

  String getShipAddress2() {
    return this.shipAddress2;
  }

  void setShipAddress2(String shipAddress2) {
    this.shipAddress2 = shipAddress2;
  }

  String getShipState() {
    return this.shipState;
  }

  void setShipState(String shipState) {
    this.shipState = shipState;
  }
}


@JsonSerializable()
class ShipmentDetails {
  String shipName;
  String shipAddress1;
  String shipCity;
  String shipPostCode;
  String shipCountry;

  ShipmentDetails(
      {@required this.shipName,
      @required this.shipAddress1,
      @required this.shipCity,
      @required this.shipPostCode,
      @required this.shipCountry});

  Map<String, dynamic> toJson() => _$ShipmentDetailsToJson(this);
  factory ShipmentDetails.fromJson(Map<String, dynamic> json) => _$ShipmentDetailsFromJson(json);

  String getShipName() {
    return this.shipName;
  }

  void setShipName(String shipName) {
    this.shipName = shipName;
  }

  String getShipAddress1() {
    return this.shipAddress1;
  }

  void setShipAddress1(String shipAddress1) {
    this.shipAddress1 = shipAddress1;
  }

  String getShipCity() {
    return this.shipCity;
  }

  void setShipCity(String shipCity) {
    this.shipCity = shipCity;
  }

  String getShipPostCode() {
    return this.shipPostCode;
  }

  void setShipPostCode(String shipPostCode) {
    this.shipPostCode = shipPostCode;
  }

  String getShipCountry() {
    return this.shipCountry;
  }

  void setShipCountry(String shipCountry) {
    this.shipCountry = shipCountry;
  }
}
